DateTime.ts

```diff
@@ -7,55 +8,14 @@ import { TimeZoneOffset } from "./TimeZoneOffset"
 export type DateTime = string
 
 export namespace DateTime {
-	function isHours(v: string): boolean {
-		return (v[0] >= "0" && v[0] <= "1" && v[1] >= "0" && v[1] <= "9") || (v[0] == "2" && v[1] >= "0" && v[1] <= "3")
-	}
-	function isMinutes(v: string): boolean {
-		return v[0] >= "0" && v[0] <= "5" && v[1] >= "0" && v[1] <= "9"
-	}
-	function isSeconds(v: string): boolean {
-		return (v[0] >= "0" && v[0] <= "5" && v[1] >= "0" && v[1] <= "9") || v == "60" || v == "61"
-	}
-	export function is(value: any | DateTime): value is DateTime {
-		// 2019-04-01T01
-		// 2019-04-01T01Z
-		// 2019-04-01T01+01:00
-		// 2019-04-01T01:11
-		// 2019-04-01T01:11Z
-		// 2019-04-01T01:11+01:00
-		// 2019-04-01T01:11:29
-		// 2019-04-01T01:11:29Z
-		// 2019-04-01T01:11:29+01:00
-		// 2019-04-01T01:11:29.000
-		// 2019-04-01T01:11:29.000Z
-		// 2019-04-01T01:11:29.000+01:00
-		// 01234567890123456789012345678
-		// 0         1         2
-		return (
-			typeof value == "string" &&
-			value.length >= 13 &&
-			value.length <= 29 &&
-			Date.is(value.substring(0, 10)) &&
-			value[10] == "T" &&
-			isHours(value.substring(11, 13)) &&
-			(value.length == 13 ||
-				TimeZoneOffset.is(value.substring(13)) ||
-				(value[13] == ":" &&
-					value.length >= 16 &&
-					isMinutes(value.substring(14, 16)) &&
-					(value.length == 16 ||
-						TimeZoneOffset.is(value.substring(16)) ||
-						(value[16] == ":" &&
-							value.length >= 19 &&
-							isSeconds(value.substring(17, 19)) &&
-							(value.length == 19 ||
-								TimeZoneOffset.is(value.substring(19)) ||
-								(value[19] == "." &&
-									value.length >= 23 &&
-									[...value.substring(20, 23)].every(c => c >= "0" && c <= "9") &&
-									(value.length == 23 || TimeZoneOffset.is(value.substring(23)))))))))
+	export const type = isly.named(
+		"isoly.DateTime",
+		isly.string<DateTime>(
+			/^((((18|19|20|21|22)[0-9]{2}-(0[13578]|1[02])-(0[1-9]|[12][0-9]|3[01]))|(18|19|20)[0-9]{2}-(0[469]|11)-(0[1-9]|[12][0-9]|30)|(18|19|20)[0-9]{2}-(02)-(0[1-9]|1[0-9]|2[0-8])|(((18|19|20)(04|08|[2468][048]|[13579][26]))|2000)-(02)-29)(T([01]\d|2[0-3]))(:[0-5]\d(:(([0-5]\d)|60)(\.\d{3}?)?)?)?)(Z|([-+](0\d|1[012])(:(00|15|30|45)?)?)?)$/
```

Comment:
Instead of this check
`Date.is(value.substring(0, 10))`
and whatever rules apply to `value.substring(10)`

Reply:
Also this regex allows `60` seconds.
Like `2020-01-02T01:02:60.000Z`

Reply:
Switched to use Date.is.
Support for leap second is required.